### PR TITLE
Pract.04: Add 12-14 | Stefka Lyaskalieva

### PR DESCRIPTION
**Pull request by stefka04**
Stefka Lyaskalieva 
2 4MI0600158

### CODE REVIEW

File: SI/Sem.04/Pract.04/Task12.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+int abs(int num)
+{
+	return num > 0 ? num : -num;
+}
+int closestPowOfTwo(int num)
+{   
+	int closest = 100000;
+	for(int i = 128; ; i *= 2)
+	{
+		if (abs(num - i) < closest)
```
Comment (GeorgiTerziev02) [inline SI/Sem.04/Pract.04/Task12.cpp:21]: The if else can be simplified to be only if

File: SI/Sem.04/Pract.04/Task13.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+using namespace std;
+bool isPrime(int num)
+{
+	int squreRoot = sqrt(num);
```
Comment (GeorgiTerziev02) [inline SI/Sem.04/Pract.04/Task13.cpp:5]: You define this variable way too early, you will use it on line 11.

File: SI/Sem.04/Pract.04/Task13.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		{
+			isPrime = false;
+			break;
+		}
+	}
+	return isPrime;
+}
+void allPrimeNums(int num)
+{
+	for (int i = 1; i < num; i++)
```
Comment (GeorgiTerziev02) [inline SI/Sem.04/Pract.04/Task13.cpp:23]: Can be optimized to skip all even numbers except 2

File: SI/Sem.04/Pract.04/Task13.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+			cout<< i <<" ";
+		}
+	}
+}
+
+bool isSumOfTwoPrimeNums(int num)
+{
+	bool isSum = false;
+	for (int i = 2; i < num; i++)
+	{
```
Comment (GeorgiTerziev02) [inline SI/Sem.04/Pract.04/Task13.cpp:36]: Can be optimized => on line 36 if(!isPrime(i)) continue. Also you can skip all the even numbers except for two.

File: SI/Sem.04/Pract.04/Task14.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using namespace std;
+int sumOfDigits(long long num)
+{
+	int digits = 0;
+	int numCopy = num;
+	int sum = 0;
+	while (numCopy != 0)
+	{
+		numCopy /= 10;
+		digits++;
```
Comment (GeorgiTerziev02) [inline SI/Sem.04/Pract.04/Task14.cpp:11]: You can directly sum the digits while you divide the number, in that way you will skip the next for loop

**Review by GeorgiTerziev02 — APPROVED** [review]
Some optimizations can be made on your solutions

**Review by Justsvetoslavov — APPROVED** [review]

**Pull request closed without merging**